### PR TITLE
feat: refactor TeamRepository to use internal types

### PR DESCRIPTION
**Pull request by hbjORbj**
# Refactor TeamRepository to use internal types instead of raw Prisma types

## Summary

This PR completes the refactoring of the `TeamRepository` pattern to cleanly separate Prisma logic from the app router layer by introducing internally-defined `select` and `where` objects. The goal is to allow upper layers (app router, services, tRPC handlers) to pass internal `TeamSelect` and `TeamFilter` objects instead of raw Prisma types, while keeping Prisma logic isolated in the data layer.

**Key Changes:**
- Created internal type definitions in `packages/lib/server/data/team/` (TeamSelect, TeamFilter)
- Added mapping helpers (`mapToPrismaSelect`, `mapToPrismaWhere`) to convert internal types to Prisma types
- Updated all TeamRepository methods to accept internal types with backward-compatible default parameters
- Added prebuilt select objects for common use cases (`teamBasicSelect`, `teamBookingPageSelect`, etc.)
- Updated UserRepository for consistency with the new pattern
- Updated test files to expect new method signatures

**Backward Compatibility:** All existing consumers should continue to work without changes due to default parameter values, but the methods now accept internal types for better separation of concerns.

## Review & Testing Checklist for Human

- [ ] **Verify type safety**: Check that internal `TeamSelect` type structure actually matches `Prisma.TeamSelect` - the mapping uses `as` casting which could hide mismatches
- [ ] **Test key consumers**: Manually test critical flows like team listing in app router (`/teams` page), tRPC team queries, and user membership lookups to ensure no regressions
- [ ] **Validate prebuilt selects**: Review complex select objects like `teamForUserMembershipSelect` and `teamWithOrganizationSettingsSelect` to ensure they contain all required fields for their intended use cases
- [ ] **Check backward compatibility**: Verify that existing method calls without explicit select parameters still work correctly (especially `findTeamsByUserId` and `findTeamWithMembers`)
- [ ] **Run comprehensive tests**: Execute `yarn test` to catch any edge cases not covered by CI, particularly around team data access patterns

**Recommended Test Plan:** 
1. Navigate to `/teams` page and verify team listing works
2. Test team creation/editing workflows 
3. Verify organization settings and member management features
4. Check tRPC team-related queries in browser dev tools

---

### Diagram

```mermaid
%%{ init : { "theme" : "default" }}%%
flowchart TD
    Apps["apps/web/app/.../teams/<br/>server-page.tsx"]:::minor-edit
    TRPC["packages/trpc/server/routers/<br/>viewer/teams/list.handler.ts"]:::minor-edit
    TeamRepo["packages/lib/server/<br/>repository/team.ts"]:::major-edit
    UserRepo["packages/lib/server/<br/>repository/user.ts"]:::minor-edit
    
    DataTypes["packages/lib/server/<br/>data/team/"]:::major-edit
    Selects["selects.ts<br/>(TeamSelect type)"]:::major-edit
    Filters["filters.ts<br/>(TeamFilter type)"]:::major-edit
    
    
    Mapper["packages/lib/server/repository/<br/>prisma-mapper.ts"]:::major-edit
    Tests["packages/lib/server/repository/<br/>TeamRepository.test.ts"]:::minor-edit
    
    Apps --> TeamRepo
    TRPC --> TeamRepo
    TeamRepo --> DataTypes
    TeamRepo --> Mapper
    UserRepo --> DataTypes
    UserRepo --> Mapper
    DataTypes --> Selects
    DataTypes --> Filters
    Tests --> TeamRepo
    
    Mapper -->|"mapToPrismaSelect<br/>mapToPrismaWhere"| TeamRepo
    
    subgraph Legend
        L1[Major Edit]:::major-edit
        L2[Minor Edit]:::minor-edit
        L3[Context/No Edit]:::context
    end

classDef major-edit fill:#90EE90
classDef minor-edit fill:#87CEEB
classDef context fill:#FFFFFF
```

### Notes

- This refactoring maintains the existing TeamRepository class structure while adding internal type support
- The approach uses type casting (`as Prisma.TeamSelect`) in the mapper, which is safe as long as internal types remain strict subsets of Prisma types
- All CI checks are passing, but manual testing of key user flows is recommended due to the wide impact of these changes
- Future consumers can now use prebuilt selects like `teamBookingPageSelect` for common use cases

**Session Info:** Requested by @hbjORbj  
**Devin Session:** https://app.devin.ai/sessions/cfcd899e40cb4267bf5ba1f84ea484ed

### CODE REVIEW

File: packages/lib/server/data/team/index.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export * from "./selects";
```
Comment (hbjORbj) [inline packages/lib/server/data/team/index.ts]: DevinAI, Let's not introduce barrel files
